c/httpfileservice.c

```diff
@@ -197,6 +197,35 @@ void deleteUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void createFileFromUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
+  if (isDir(absolutePath)) {
+      int folderNameLen = strlen(absolutePath);
+      if(folderNameLen == 0 || absolutePath == NULL) {
+         respondWithJsonError(response, "Failed to idenity the folder pointed", 400, "Bad Request");
+      }
+      int slashPos = lastIndexOf(absolutePath, folderNameLen, '/');
+      char *tarFileName = (slashPos == -1) ? absolutePath : absolutePath + slashPos + 1;
+      char finalFileName[strlen(tarFileName) + strlen(".tar")];
```

Comment:
This buffer is not big enough. You'll get a buffer overflow.

Reply:
so  @ifakhrutdinov  what would be the suggestion here ?

Reply:
A simple thing to do is use `sprintf` of the values you want into a buffer of length https://github.com/zowe/zowe-common-c/blob/master/h/unixfile.h#L472

Reply:
sure